Source/JavaScriptCore/wasm/WasmCallingConvention.h

```diff
@@ -117,34 +135,55 @@ class WasmCallingConvention {
     WTF_MAKE_NONCOPYABLE(WasmCallingConvention);
 
 private:
+    template <typename RegType>
```

Comment:
Can we just make a separate arm marshaller?

Reply:
Yah, that sounds simpler.

Reply:
To be clear, would you prefer a separate `WasmCallingConventionArmThumb2&` class (like we do for `CCallingConventionArmThumb2` in the same file) or a different (`CPU(ARM_THUMB2)`) implementation of the methods in the same class. The former would ensure maximal separation, but then we can't have the callers be architecture-agnostic (at least, not without defining an abstract class interface).

Reply:
I'm not sure a separate `WasmCallingConvention` would be simpler.  It looks like there's a lot of code in common right now.

Reply:
Kk, will upload a new version which only makes minimal changes to this file, as in an earlier iteration.